packages/databyss-ui/modules/SourcesValueList/SourceModal.js

```diff
@@ -0,0 +1,46 @@
+import React, { useState } from 'react'
+import { useSourceContext } from '@databyss-org/services/sources/SourceProvider'
+import { ModalWindow } from '@databyss-org/ui/primitives'
+import { hideModal } from '@databyss-org/ui/components/Navigation/NavigationProvider/actions'
+import { useNavigationContext } from '@databyss-org/ui/components/Navigation/NavigationProvider/NavigationProvider'
+import SourcesValueList from './SourcesValueList'
+
+const SourceModal = ({ sourceId, visible, onUpdateSource }) => {
+  const { setSource } = useSourceContext()
+  const [values, setValues] = useState(null)
+  const [, dispatchNav] = useNavigationContext()
+
+  const onChange = _values => {
+    // update internal state
+    setValues(_values)
+    // if (values) {
+    //   // updates in source provider
+    //   setSource(_values)
+    // }
+  }
+
+  const onDismiss = () => {
+    if (values) {
+      // updates in source provider
+      setSource(values)
```

Comment:
It looks like this only updates the source when the modal is dismissed, but we want them updating whenever a field blurs

Reply:
Updating the sourcesProvider `onBlur` causes the value list to re-render and loose focus. we can discuss this on our next chat